DQMOffline/Trigger/plugins/LepHTMonitor.cc

```diff
@@ -0,0 +1,488 @@
+    }
+  }
+
+  //Beam Spot
+  edm::Handle<reco::BeamSpot> BeamSpot;
+  if(theBeamSpotTag_.label() != ""){
+    e.getByToken(theBeamSpot_, BeamSpot);
+    if( !BeamSpot.isValid() ){
+      edm::LogWarning("LepHTMonitor")
+        << "Invalid BeamSpot: " << theBeamSpotTag_.label() << '\n';
+    }
+  }
+
+  //MET
+  edm::Handle<reco::PFMETCollection> pfMETCollection;
+  if(thePfMETTag_.label() != ""){
+    e.getByToken(thePfMETCollection_, pfMETCollection);
+    if( !pfMETCollection.isValid() ){
+      edm::LogWarning("LepHTMonitor")
+        << "Invalid PFMETCollection: " << thePfMETTag_.label() << '\n';
+    }
+  }
+
+  //Jets
+  edm::Handle<reco::PFJetCollection> pfJetCollection;
+  if(thePfJetTag_.label() != ""){
+    e.getByToken (thePfJetCollection_,pfJetCollection);
+    if( !pfJetCollection.isValid() ){
+      edm::LogWarning("LepHTMonitor")
+        << "Invalid PFJetCollection: " << thePfJetTag_.label() << '\n';
+    }
+  }
+
+  //Electron
+  edm::Handle<edm::View<reco::GsfElectron> > ElectronCollection;
+  if(theElectronTag_.label() != ""){
+    e.getByToken (theElectronCollection_, ElectronCollection);
+    if( !ElectronCollection.isValid() ){
+      edm::LogWarning("LepHTMonitor")
+        << "Invalid GsfElectronCollection: " << theElectronTag_.label() << '\n';
+    }
+  }
+  
+  //Muon
+  edm::Handle<reco::MuonCollection> MuonCollection;
+  if(theMuonTag_.label() != ""){
+    e.getByToken (theMuonCollection_, MuonCollection);
+    if( !MuonCollection.isValid() ){
+      edm::LogWarning("LepHTMonitor")
+        << "Invalid MuonCollection: " << theMuonTag_.label() << '\n';
+    }
+  }
+
+  //Get offline HT
+  double pfHT = -1.0;
+  if(pfJetCollection.isValid()){
+    pfHT=0.0;
+    for(const auto &pfjet: *pfJetCollection){
+      if(pfjet.pt() < jetPtCut_) continue;
+      if(fabs(pfjet.eta()) > jetEtaCut_) continue;
+      pfHT += pfjet.pt();
+    }
+  }
+
+  //Get offline MET
+  double pfMET = -1.0;
+  if(pfMETCollection.isValid() && pfMETCollection->size()){
+    pfMET = pfMETCollection->front().et();
+  }
+
+
+  //Find offline leptons and keep track of pt,eta of leading and trailing leptons
+  double lep_max_pt = -1.0;
+  double lep_eta=0;
+  double lep_phi=0;
+  double trailing_ele_eta=0;
+  double trailing_ele_phi=0;
+  double trailing_mu_eta=0;
+  double trailing_mu_phi=0;
+  double min_ele_pt= -1.0;
+  double min_mu_pt=-1.0;
+  int nels=0;
+  int nmus=0;
+  if(VertexCollection.isValid() && VertexCollection->size()){//for quality checks
+    //Try to find a reco electron
+    if(ElectronCollection.isValid()
+       && ConversionCollection.isValid()
+       && BeamSpot.isValid()
+       && ele_id_decisions.isValid()){
+      size_t index=0;
+      for(const auto &electron: *ElectronCollection){
+	const auto el = ElectronCollection->ptrAt(index);
+	bool pass_id = (*ele_id_decisions)[el];
+        if(isGood(electron, VertexCollection->front().position(),
```

Comment:
it also doesn't help to mix tabs and spaces - they look OK on the terminal but not so much on GitHub